diff --git a/ldap3/abstract/entry.py b/ldap3/abstract/entry.py
index ccbe932..199f8ed 100644
--- a/ldap3/abstract/entry.py
+++ b/ldap3/abstract/entry.py
@@ -28,7 +28,7 @@ import json
 try:
     from collections import OrderedDict
 except ImportError:
-    from ..utils.ordDict import OrderedDict  # for Python 2.6
+    from ordereddict import OrderedDict  # for Python 2.6
 
 from os import linesep
 
iff --git a/setup.py b/setup.py
index 313db60..3ad8915 100644
--- a/setup.py
+++ b/setup.py
@@ -23,6 +23,7 @@
 # along with ldap3 in the COPYING and COPYING.LESSER files.
 # If not, see <http://www.gnu.org/licenses/>.
 
+import sys
 import os
 from setuptools import setup
 from json import load
@@ -53,9 +54,13 @@ packages=['ldap3',
           'ldap3.extend.novell',
           'ldap3.extend.microsoft',
           'ldap3.extend.standard']
+install_requires=[i.strip() for i in open('requirements.txt').readlines()],
+if sys.version_info < (2, 7):
+    install_requires.append('ordereddict')
 
 setup_kwargs = {'packages': packages,
-                'package_dir': {'': package_folder}}
+                'package_dir': {'': package_folder},
+                'install_requires': install_requires}
 
 try:
     from Cython.Build import cythonize
@@ -148,7 +153,6 @@ if 'LDAP3_CYTHON_COMPILE' in os.environ and HAS_CYTHON is True:
 
 setup(name=package_name,
       version=version,
-      install_requires=[i.strip() for i in open('requirements.txt').readlines()],
       license=license,
       author=author,
       author_email=email,
