diff --git a/ldap3/abstract/entry.py b/ldap3/abstract/entry.py
index ccbe932..199f8ed 100644
--- a/ldap3/abstract/entry.py
+++ b/ldap3/abstract/entry.py
@@ -28,7 +28,7 @@ import json
 try:
     from collections import OrderedDict
 except ImportError:
-    from ..utils.ordDict import OrderedDict  # for Python 2.6
+    from ordereddict import OrderedDict  # for Python 2.6
 
 from os import linesep
 
diff --git a/setup.py b/setup.py
index 3ba012a..a85901e 100644
--- a/setup.py
+++ b/setup.py
@@ -23,6 +23,8 @@
 # along with ldap3 in the COPYING and COPYING.LESSER files.
 # If not, see <http://www.gnu.org/licenses/>.
 
+import sys
+
 from setuptools import setup
 from json import load
 
@@ -39,6 +41,10 @@ status = str(version_dict['status'])
 
 long_description = str(open('README.rst').read())
 
+install_requires = [i.strip() for i in open('requirements.txt').readlines()]
+if sys.version_info < (2, 7):
+    install_requires.append('ordereddict')
+
 setup(name=package_name,
       version=version,
       packages=['ldap3',
@@ -56,7 +62,7 @@ setup(name=package_name,
                 'ldap3.extend.microsoft',
                 'ldap3.extend.standard'],
       package_dir={'': package_folder},
-      install_requires=[i.strip() for i in open('requirements.txt').readlines()],
+      install_requires=install_requires,
       license=license,
       author=author,
       author_email=email,
